File: android/app/src/main/kotlin/com/goalmaker/app/application/planning/ProjectDraft.kt
package com.goalmaker.app.application.planning

/** What a new project or an edit says (docs/projects.md). */
data class ProjectDraft(
    val name: String,
    val description: String = "",
    val areaId: String? = null,
    val status: String = ProjectRules.ACTIVE,
    val repositoryUrl: String? = null,
    val localFolder: String? = null,
    val notes: String = "",
)

File: windows/src/GoalMaker.Core/Planning/ProjectDraft.cs
namespace GoalMaker.Core.Planning;

/// <summary>What a new project or an edit says (docs/projects.md).</summary>
public sealed record ProjectDraft(string Name)
{
    public string Description { get; init; } = string.Empty;

    public string? AreaId { get; init; }

    public string Status { get; init; } = ProjectRules.Active;

    public string? RepositoryUrl { get; init; }

    public string? LocalFolder { get; init; }

    public string Notes { get; init; } = string.Empty;
}
